internal/tablemap.go

```diff
@@ -127,3 +127,43 @@ func GetSpannerCols(conv *Conv, srcTable string, srcCols []string) ([]string, er
 	}
 	return spCols, nil
 }
+
+// GetSpannerKeyName maps source foreign key name to
+// legal Spanner foreign key name
+// If the srcKeyName is empty string we can just return
+// empty string without error.
+// If the srcKeyName is not empty we need to make sure
+// of the following things:
+// a) the new foreign key name is legal
+// b) the new foreign key name doesn't clash with other Spanner
+//    foreignkey names
+// c) we consistently return the same key name.
+func GetSpannerKeyName(conv *Conv, srcKeyName string, schemaForeignKeys map[string]bool) (string, error) {
```

Comment:
We don't use conv -- probably delete it?

As you noted elsewhere, we could put the schemaForeignKey map in conv, but it's probably not worth doing that since we only use the map very locally within toddl.

Reply:
updated